### ISSUE TITLE
running quit() doesn't reset the terminal

### ISSUE THREAD

**Issue by alexanderkjall** [gyscos/cursive#415]
First: is this a bug report? A suggestion? Or asking for help?

A bug report I think, or maybe asking for help :)

# Problem description
I have a quit button in a ripasso, and in the new user wizard there is a quit button that I have implemented like this:

```
        .button(super::CATALOG.gettext("Cancel"), |s| {
            s.quit();
            std::process::exit(0);
        })
```

This doesn't reset the terminal for the user and it looks like this:

![fil](https://user-images.githubusercontent.com/647710/71543055-a8999400-296e-11ea-8f9b-194798734527.png)

If possible code, or pseudo-code to reproduce the problem / present an idea.

See above.

# Environment

* Operating system: ubuntu, gnome-terminal
* Backend used: ncurses (the default one), pancurses, termion, ...
  * If using ncurses on linux, ncurses version? #define NCURSES_VERSION "6.1"
* Current locale (run `locale` in a terminal).
LANG=sv_SE.UTF-8
LANGUAGE=en
LC_CTYPE="sv_SE.UTF-8"
LC_NUMERIC="sv_SE.UTF-8"
LC_TIME="sv_SE.UTF-8"
LC_COLLATE="sv_SE.UTF-8"
LC_MONETARY="sv_SE.UTF-8"
LC_MESSAGES="sv_SE.UTF-8"
LC_PAPER="sv_SE.UTF-8"
LC_NAME="sv_SE.UTF-8"
LC_ADDRESS="sv_SE.UTF-8"
LC_TELEPHONE="sv_SE.UTF-8"
LC_MEASUREMENT="sv_SE.UTF-8"
LC_IDENTIFICATION="sv_SE.UTF-8"
LC_ALL=

* Cursive version (from crates.io, from git, ...) 0.13.0

**Comment by gyscos** [comment]
Hi, and thanks for the report!

Indeed, the terminal is only dropped when the `Cursive` object is dropped.

There is a planned redesign to change that and only initialize/reset the terminal in the `run()` method. This way, exiting the event loop would reset the terminal. We just need a bit of refactor to handle the partially initialized backend.

**Comment by Inc0ns1st3nt** [comment]
Yeah I am having some similar issue, it was working when I wrapped my `Controller` Struct, which owns the `Cursive` Object, in `AtomicPtr`, which is unsafe.
Then I learnt from [capture variable must satisfy static lifetime in closure](https://stackoverflow.com/questions/46039441/type-must-satisfy-the-static-lifetime), and I used `Rc<RefCell<>>` to wrap my `Controller`.
Which might be causing me to have the `Cursive` object still being referred from my `Rc<RefCell<Controller>>` somehow, after `s.quit()`

```
fn main() {
    let controller = Controller::alloc();

    controller.borrow_mut()
        // .init()
        .siv.run();  // Starts the event loop.
}
```
